tests/test_categorical_imputer.py

```diff
@@ -129,3 +129,25 @@ def test_not_fitted():
     imp = CategoricalImputer()
     with pytest.raises(NotFittedError):
         imp.transform(np.array(['a', 'b', 'b', None]))
+
+
+@pytest.mark.parametrize('input_type', ['np', 'pd'])
+@pytest.mark.parametrize('replacement_value', ['a', 'c'])
+def test_custom_replacement(replacement_value, input_type):
+    """
+    If replacement != 'mode', impute with that value instead of mode
+    """
+    data = ['a', np.nan, 'b', 'b']
+
+    if input_type == 'pd':
+        X = pd.Series(data)
+    else:
+        X = np.asarray(data, dtype=object)
+
+    Xc = X.copy()
+
+    Xt = CategoricalImputer(replacement=replacement_value).fit_transform(X)
+
+    assert (np.asarray(X) == np.asarray(Xc)).all()
+    assert type(Xt) == np.ndarray
```

Comment:
Better `isinstance(Xt, np.ndarray)` since it's not usually recommended to test for types explicitly in Python.

Reply:
I copied this from one of the other test cases - is it worth changing that one as well?

Reply:
Yes, please.

Reply:
Fixed in the latest commit.